scripts/build_osm_network.py

```diff
@@ -0,0 +1,1051 @@
+# -*- coding: utf-8 -*-
+# SPDX-FileCopyrightText: : 2020-2024 The PyPSA-Eur and PyPSA-Earth Authors
+#
+# SPDX-License-Identifier: MIT
+
+import logging
+import os
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+from _benchmark import memory_logger
+from _helpers import configure_logging, set_scenario_config
+from shapely.geometry import LineString, Point
+from shapely.ops import linemerge, split
+from tqdm import tqdm
+
+logger = logging.getLogger(__name__)
+
+# list of recognised nan values (NA and na excluded as may be confused with Namibia 2-letter country code)
+NA_VALUES = ["NULL", "", "N/A", "NAN", "NaN", "nan", "Nan", "n/a", "null"]
+
+
+def read_csv_nafix(file, **kwargs):
+    "Function to open a csv as pandas file and standardize the na value"
+    if "keep_default_na" not in kwargs:
+        kwargs["keep_default_na"] = False
+    if "na_values" not in kwargs:
+        kwargs["na_values"] = NA_VALUES
+
+    if os.stat(file).st_size > 0:
+        return pd.read_csv(file, **kwargs)
+    else:
+        return pd.DataFrame()
+
+
+def save_to_geojson(df, fn):
+    """
+    Save a (Geo)DataFrame to a GeoJSON file.
+
+    Parameters:
+    - df: The (Geo)DataFrame to be saved.
+    - fn: The filename (including the path) of the output GeoJSON file.
+
+    Returns:
+    None
+    """
+    if os.path.exists(fn):
+        os.unlink(fn)  # remove file if it exists
+
+    # save file if the (Geo)DataFrame is non-empty
+    if df.empty:
+        # create empty file to avoid issues with snakemake
+        with open(fn, "w") as fp:
+            pass
+    else:
+        # save file
+        df.to_file(fn, driver="GeoJSON")
```

Comment:
Let's explore if all calls of `save_to_geojson` can be replaced by `df.to_file(fn)`. Even the driver can be left out as it is inferred from file ending.

Reply:
done, removed and simplified